tools/test-reader/cmd/read_tests.go

```diff
@@ -0,0 +1,63 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/magic-modules/tools/test-reader/reader"
+	"github.com/spf13/cobra"
+)
+
+const readTestsDesc = "Run the missing test detector using the given services directory"
+
+type readTestsOptions struct {
+	rootOptions *rootOptions
+	testPrefix  string
+}
+
+func newReadTestsCmd(rootOptions *rootOptions) *cobra.Command {
```

Comment:
Just to make sure I understand the purpose of this PR correctly. 
1. we moved the test reader away from the diff processor 
2. We implement the feature to run the missing test detector for the given services directory. This is not actually being called in our CI at this moment, is it?

Reply:
1. yes
2. this is only implemented as a new command for the test reader. This behavior is used by the missing test detector, but that's still part of the diff-processor. The missing test detector uses the output from `reader.ReadAllTests`, then filters it by what fields / resources were modified according to the diff.

I wanted to split this out because it makes it a lot easier to run the test reader on its own and debug why it's having trouble with a specific test file.

Reply:
Makes sense. Thanks for clarifying!